docs/painless/painless-guide/painless-execute-script.asciidoc

```diff
@@ -154,13 +255,566 @@ POST /_scripts/painless/_execute
+Several options are available for using
+<<painless-datetime,using datetime in Painless>>. In this example, you'll
+estimate when a particular author starting writing a book based on its release
+date and the writing speed of that author. The example makes some assumptions,
+but shows to write a script that operates on a date while incorporating
+additional information.
+
+Add the following fields to your index mapping to get started:
+
+[source,console]
+----
+PUT /my-index-000001
+{
+  "mappings": {
+    "properties": {
+      "name": {
+        "type": "keyword"
+      },
+      "author": {
+        "type": "keyword"
+      },
+      "release_date": {
+        "type": "date"
+      },
+      "page_count": {
+        "type": "long"
+      }
+    }
+  }
+}
+----
+
+The following script adds a variable named `pageTime` and makes an incredible
+assumption that this is an approximate time required to write a complete page
+for a book.
+
+The script then retrieves the `author` and makes another fantastic assumption
+to either divide or multiply the `pageTime` value based on the author's
+perceived writing speed (yet another wild assumption).
+
+The script subtracts the release date value (in milliseconds) from the
+calculation of `pageTime` times the `page_count` to determine approximately
+(based on numerous assumptions) when the author began writing the book.
+
+[source,console]
+----
+POST /_scripts/painless/_execute
+{
+  "script": {
+    "source": """
+      String author = doc['author'].value;
+      long pageTime = 28800000;  <1>
+      if (author == 'Robert A. Heinlein') {
+        pageTime /= 2;           <2>
+      } else if (author == 'Alastair Reynolds') {
+        pageTime *= 2;           <3>
+      }
+      emit(doc['release_date'].value.toInstant().toEpochMilli() - pageTime * doc['page_count'].value);
+    """
+  },
+  "context": "date_field",
+  "context_setup": {
+    "index": "my-index-000001",
+    "document": {
+      "name": "Revelation Space",
+      "author": "Alastair Reynolds",
+      "release_date": "2000-03-15",
+      "page_count": 585
+    }
+  }
+}
+----
+//TEST[continued]
+<1> Eight hours, represented in milliseconds
+<2> Incredibly fast writing from Robert A. Heinlein
+<3> Alastair Reynolds writes space operas at a much slower speed
+
+In this case, the author is Alastair Reynolds. Based on a release date of
+`2000-03-15`, the script calculates that the author started writing
+`Revelation Space` on 19 February 1999. Writing a 585 page book in just over one
+year is pretty impressive!
+
+[source,console-result]
+----
+{
+  "result" : [
+    "1999-02-19T00:00:00.000Z"
+  ]
+}
+----
+
+[[painless-runtime-double]]
+===== `double_field`
+Use the `double_field` context for {ref}/number.html[numeric data] of type
+`double`. For example, let's say you have sensor data that includes a `voltage`
+field with values like 5.6. You need to multiply this value, but only for
+sensors that match a specific model number.
```

Comment:
Maybe "But you find out after indexing millions of documents that sensor with the model number `QVKC92Q` underreport their voltage by a factor of `1.7`. Rather than rebuild your data you fix it with a runtime field."

Reply:
Love this, and will include it.